pennylane_qiskit/qiskit_device2.py

```diff
@@ -447,12 +472,15 @@ def execute_circuits(session):
 
     def _execute_sampler(self, circuit, session):
         """Execution for the Sampler primitive"""
-
         qcirc = [circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)]
         sampler = Sampler(session=session)
         compiled_circuits = self.compile_circuits(qcirc)
+        sampler.options.update(**self._kwargs)
 
-        result = sampler.run(compiled_circuits).result()[0]
+        result = sampler.run(
+            compiled_circuits,
+            shots=circuit.shots.total_shots if circuit.shots.total_shots else None,
```

Comment:
Shots defined in `circuit` take precedence over shots defined on device initialization.

Reply:
Nice!